src/js/components/CollapsibleErrorMessage.js

```diff
@@ -0,0 +1,227 @@
+import Icon from './Icon';
+
+import React from 'react';
+import classNames from 'classnames';
+
+/**
+ * Methods to bind in 'this' context
+ */
+const METHODS_TO_BIND = [
+  'toggleExpanded'
+];
+
+/**
+ * An error bar component commonly used in the ServiceForm and the JobsForm
+ * that contains collapsible error messages.
+ *
+ * @example <caption>How to use the CollapsibleErrorMessage</caption>
+ *   <CollapsibleErrorMessage
+ *
+ *        className="error-for-modal"
+ *
+ *        details={[ 'an', 'array', 'of', 'detailed errors' ]}
+ *        message="The message to display"
+ *
+ *        onToggle={function(isToggled) {
+ *          // Do something
+ *        }}
+ *
+ *      />
+ */
+class CollapsibleErrorMessage extends React.Component {
+
+  /**
+   * Initialize superclass
+   */
+  constructor() {
+    super(...arguments);
+
+    // Initial state
+    this.state = {
+      expanded: false
+    };
+
+    // Bind methods in context
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+
+  }
+
+  /**
+   * Set the default expandedg from the properties
+   */
+  componentWillMount() {
+    this.setState({
+      expanded: this.props.expanded
+    });
+  }
+
+  /**
+   * Toggle the expanded state of the collapsible error bar
+   */
+  toggleExpanded() {
+    let expanded = !this.state.expanded;
+    this.setState({expanded});
+
+    // Call onToggle callback
+    this.props.onToggle(expanded);
+  }
+
+  /**
+   * [Fragment] The show more/less link
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getShowDetailsLink() {
+    let {message, details} = this.props;
+    let isDetailed = message && details && details.length;
+
+    // Check if we must not show the detail link at all
+    if (!isDetailed) {
+      return null;
+    }
+
+    // Get text
+    let moreLess = 'more';
+    if (this.state.expanded) {
+      moreLess = 'less';
+    }
+
+    // Render
+    return (
+        <span
+          className="collapsible-toggle-label  clickable"
+          onClick={this.toggleExpanded} >
+          (Show {moreLess})
+        </span>
+      );
+
+  }
+
+  /**
+   * [Fragment] The list of errors in the details
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getDetailsListItems() {
+    let {details} = this.props;
+
+    return details.map(function (message) {
+      let msg = message.toString();
+
+      return (
+          <li key={msg}>{msg}</li>
```

Comment:
Let's not use `msg` as our key. If your errors are showing an identical string, this will mess up with React's rendering efforts.

In this case, the easiest will be:
```
return details.map(function (message, i) {
// ...
return  <li key={i}>{msg}</li>
```